### DISCUSSION TITLE
zingg-0.3.4-SNAPSHOT-spark-3.1.2

### DISCUSSION THREAD

**Discussion by sonalgoyal** [zinggAI/zingg#419] [Announcements]
Release with 
a. Python interface
b. Stopwords
c. New match types - email, pin etc
d. Support for integers, doubles and dates

<hr /><em>This discussion was created from the release <a href='https://github.com/zinggAI/zingg/releases/tag/'>zingg-0.3.4-SNAPSHOT-spark-3.1.2</a>.</em>

**Comment by jinesh-vds** [comment]
@sonalgoyal Following the (https://www.zingg.ai/documentation-article/identity-resolution-on-databricks-for-customer-360) blog and when running the jar in the end for trainMatch phase. It is failing with below error.

props={location=models/100/trainingData//marked/} is not correct path. It seems when this path is getting generated using the jar code is adding extra slash between trainingData and marked.

**Comment by sonalgoyal** [comment]
check https://github.com/zinggAI/zingg/blob/main/examples/databricks/FebrlExample.ipynb using Zingg 0.4.0 and DBR 14.2

**Comment by jinesh-vds** [comment]
@sonalgoyal Tried running the notebook with DBR 14.2 and the notebooks is failing where the arguments object to be created with the below error.- 

TypeError: 'JavaPackage' object is not callable
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
File <command-988354905897265>, line 2
      1 #build the arguments for zingg
----> 2 args = Arguments()
      3 # Set the modelid and the zingg dir. You can use this as is
      4 args.setModelId(modelId)

**Reply by sonalgoyal** [reply]
Have you set the Zingg jar as a library on the cluster?

**Reply by jinesh-vds** [reply]
@sonalgoyal After setting the Zingg jar as library not getting the error which I mentioned in the above comment but now getting the below error in cell 25 while trying to run the 'findTrainingData'. also commands are missing in cell 10 and 11 in notebook. Kindly advise on that as well.

Error:

Py4JJavaError                             Traceback (most recent call last)
File <command-988354905897275>, line 5
      3 #Zingg execution for the given phase
      4 zingg = ZinggWithSpark(args, options)
----> 5 zingg.initAndExecute()

File /local_disk0/.ephemeral_nfs/envs/pythonEnv-4f0b1aa6-f6e0-4f4f-bc11-9720d8a947d7/lib/python3.10/site-packages/zingg/client.py:144, in Zingg.initAndExecute(self)
    142         self.client.execute()
    143 else:
--> 144     self.client.execute()

**Reply by mandeepsb** [reply]
@sonalgoyal @jinesh-vds - I am getting this same error when I try to run on Microsoft Fabric. Were you able to find a fix for this issue?

<img width="1023" height="655" alt="Screenshot 2025-10-02 160413" src="https://github.com/user-attachments/assets/1a8083ac-7cd5-4ea2-a0a2-94ca095766fa" />

**Comment by sonalgoyal** [comment]
Java package is not callable happens when the Zingg python api can not find the Zingg jar. Please check if the jar is part of the cluster library
